internal/testutils/nettest/free_port.go

```diff
@@ -0,0 +1,38 @@
+package nettest
+
+import "net"
+
+func getClosedTCPAddr() (*net.TCPAddr, error) {
+	address, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, err
+	}
+
+	listener, err := net.ListenTCP("tcp", address)
+	if err != nil {
+		return nil, err
+	}
+	defer listener.Close()
+
+	return listener.Addr().(*net.TCPAddr), nil
+}
+
+// MustGetFreeHostPort returns a TCP host:port that is free for unit tests
+// that cannot use port 0.
+func MustGetFreeHostPort() string {
```

Comment:
@abg and @breerly rejected me exposing this before, but Id want to do it again. However, this doesn't fully fix the problem - you need to retain the listener connection, because when you close, someone else can come in and use that port (i.e. there's a race condition here).

Going further, we need a way to do this across processes, i.e. lock down a port to use in two separate processes (for examples testing). I'd use a lock file but as I said this whole thing was shot down.